ReactNativeClient/lib/services/searchengine/SearchEngine.js

```diff
@@ -635,7 +651,15 @@ class SearchEngine {
 		// If preferredSearchType is "fts" we auto-detect anyway
 		// because it's not always supported.
 
-		const st = scriptType(query);
+		let allTerms = [];
+		try {
+			allTerms = filterParser(query);
+		} catch (error) {
+			console.warn(error);
```

Comment:
Should we proceed with the rest of the code if there was an error here? What kind of error be thrown in this case?

Reply:
You're right, we could throw an error here and exit from this function. But there needs to be a catch statement higher up.

Errors would be invalid filters like `-notebook:abc`, `type:blah` , `foo:bar` etc.

Reply:
Yes it seems that getting the error displayed somewhere in the UI wouldn't be that easy, so let's keep it as it is now. A warning is a good start and there's the doc to explain how the query should be formatted.